src/runtime/micro/target_data_layout_encoder.h

```diff
@@ -0,0 +1,175 @@
+/*!
+ *  Copyright (c) 2019 by Contributors
+ * \file target_data_layout_encoder.h
+ * \brief uTVM data layout encoder
+ */
+#ifndef TVM_RUNTIME_MICRO_TARGET_DATA_LAYOUT_ENCODER_H_
+#define TVM_RUNTIME_MICRO_TARGET_DATA_LAYOUT_ENCODER_H_
+
+#include <dmlc/memory_io.h>
+
+#include <algorithm>
+#include <cstring>
+#include <iomanip>
```

Comment:
Check the headers and see if we really need them (put them in cc files which use them instead of the header). We want to minimize the header dependencies

Reply:
will do, but do we not have a linter that checks for this?

Reply:
Unfortunately we do not